drivers/wdm/audio/backpln/portcls/irpstream.cpp

```diff
@@ -779,23 +780,50 @@ CIrpQueue::GetCurrentIrpOffset()
 BOOLEAN
 NTAPI
 CIrpQueue::GetAcquiredTagRange(
-    IN PVOID * FirstTag,
-    IN PVOID * LastTag)
+    OUT PVOID * FirstTag,
+    OUT PVOID * LastTag)
 {
     KIRQL OldLevel;
     BOOLEAN Ret = FALSE;
-    //PIRP Irp;
-    //PLIST_ENTRY CurEntry;
-    //PKSSTREAM_DATA StreamData;
+    PIRP Irp;
+    PLIST_ENTRY CurEntry;
+    PKSSTREAM_DATA StreamData;
 
     // lock list
     KeAcquireSpinLock(&m_IrpListLock, &OldLevel);
 
-    // initialize to zero
-    *FirstTag = NULL;
-    *LastTag = NULL;
+    CurEntry = m_IrpList.Flink;
+
+    while (CurEntry != &m_IrpList)
```

Comment:
This is C++, don't we have some kind of an iterator for lists?

Reply:
:thinking: I don't remember anything like that.